### ISSUE TITLE
custom-server example build script not working with next.js 14

### ISSUE THREAD

**Issue by jeremiasjutz** [payloadcms/payload#4119]
### Describe the Bug

When cloning the custom-server example and upgrading the next.js version to 14, the `build` fails. It looks as if it is trying to find _error files from the pages directory. But as I am using the app router, this is a blocking issue.

```sh
ERROR (payload): Export encountered errors on following paths:
        /_error: /404
        /_error: /500
        /_not-found
        /page: /
```

My current workaround is changing the NEXT_BUILD step in the server.ts to the following:
```ts
if (process.env.NEXT_BUILD) {
  app.listen(PORT, async () => {
    payload.logger.info(`Next.js is now building...`);
    const nextBuildConfig = {
      dir: path.join(__dirname, ".."),
      reactProductionProfiling: undefined,
      debugOutput: undefined,
      runLint: undefined,
      noMangling: undefined,
      appDirOnly: undefined,
      turboNextBuild: undefined,
      turboNextBuildRoot: undefined,
      buildMode: "experimental-compile",
    };
    try {
      // @ts-expect-error
      await nextBuild(...Object.values(nextBuildConfig));
    } catch (error) {
      payload.logger.error(error);
    }
    process.exit();
  });

  return;
}
```

The only thing I am actually changing is using the "experimental-compile". Is this a viable option or is there a better solution?

### To Reproduce

Clone the custom-server example. Upgrade to next.js 14. Run the `build` script. Error happens at `await nextBuild(...)`.

### Payload Version

2.1.0

### Adapters and Plugins

db-postgres

**Comment by PatrikKozak** [comment]
Hey @jeremiasjutz - I just went ahead and upgraded to next.js 14 in the custom-server example as well and was able to build successfully. What steps did you take to upgrading to next.js 14?

I went ahead and followed the steps in the next docs [here](https://nextjs.org/docs/pages/building-your-application/upgrading/version-14). 

Which involved also updating a couple other dependencies / devDependencies to latest - e.g. `react`, `react-dom`, `eslint-config-next`, `@types/react` and `@types/react-dom`.

**Comment by jeremiasjutz** [comment]
Hi @PatrikKozak Thanks for your response. Yes I updated the project according to the docs. Unfortunately I still get weird errors at build which causes it to fail (unless I add the described function args above)
Here is a snippet of the errors I get: 
```sh
Generating static pages (0/4)  [    ]Error: <Html> should not be imported outside of pages/_document.
...
Error occurred prerendering page "/404". Read more: https://nextjs.org/docs/messages/prerender-error
Error: <Html> should not be imported outside of pages/_document.
Read more: https://nextjs.org/docs/messages/no-document-import-in-page
...
Error: <Html> should not be imported outside of pages/_document.
Read more: https://nextjs.org/docs/messages/no-document-import-in-page
...
Error occurred prerendering page "/500". Read more: https://nextjs.org/docs/messages/prerender-error
Error: <Html> should not be imported outside of pages/_document.
Read more: https://nextjs.org/docs/messages/no-document-import-in-page
...
[10:40:07] ERROR (payload): Export encountered errors on following paths:
        /_error: /404
        /_error: /500
        /_not-found
        /page: /
    err: {
      "type": "ExportError",
      "message": "Export encountered errors on following paths:\n\t/_error: /404\n\t/_error: /500\n\t/_not-found\n\t/page: /",
      "stack":
          Error: Export encountered errors on following paths:
                /_error: /404
                /_error: /500
                /_not-found
                /page: /
...
        "code": "NEXT_EXPORT_ERROR"
...
```

**Comment by jeremiasjutz** [comment]
Update! The error seems to be related to @payloadcms/bundler-vite. When using bundler-webpack, the build works as expected. Is it required to use bundler-webpack in this case?

**Comment by PatrikKozak** [comment]
@jeremiasjutz Yup, I was able to reproduce this as well using `@payloadcms/bundler-vite`. 

On initial glance, these errors seem to be issues with Next. My suspicions come from these issues that I just came across:
- https://github.com/vercel/next.js/issues/58576
- https://github.com/vercel/next.js/issues/52158

I will continue looking into this and get back to you asap!

**Comment by PatrikKozak** [comment]
Hey @jeremiasjutz - so the issue stems from the fact that NextJS uses webpack under the hood so in this case the webpack bundler is required. 

Going to close this issue for now but if there is anything else I can assist with, let me know.

**Comment by ashwin-pillai-dev** [comment]
Hey @PatrikKozak  - I created a new Next app using the create-next-app command and tried building it using npm run build and still got the same issue.

Steps to reproduce:
1. create a new next app using npx create-next-app@latest
2.npm run build

I'm on node version v19.9.0

**Comment by pvaladez** [comment]
In case anyone else has a similar issue, I had modified the `/app/(pages)/[...slug]/page.tsx` file and the Pages collection so that you can use an empty slug for the home page.  The payload template defaults the slug to 'home', but then nothing comes up at the root of your site... not sure if there is a better way to solve that issue.  Anyway, because the slug is just an empty string, my `generateStaticParams()` function was returning `{ slug: [''] }`, which caused next.js to report the prerender / `"code": "NEXT_EXPORT_ERROR"` error.  Changing `generateStaticParams()` to return `{ slug: ['/'] }` fixed the problem for me.
